.github/workflows/ci.yml

```diff
@@ -1,5 +1,8 @@
 name: test
-on: [ push ]
+on:
+  pull_request:
+    types: [ opened ]
+  push:
```

Comment:
Looks fine to me, but probably we should consider only pull request to main branch and also the reopened and synchronize. 

Regarding the push method it could be fine to me if we can add a sort of wildcard that will enable the ci to any branch that receive a push. 

What's your thought?